### PR TITLE
Fixes filter application

### PR DESCRIPTION
**Pull request by therobbiedavis**
#Added
- Added: Added a new API for Tachiyomi Progress Sync (Tachiyomi Sync #964 )

# Fixed
- Fixed: Fixed an issue where pagination would not reset on filter application (develop)
- Fixed: When no user exists for an api key, throw a 401.

### CODE REVIEW

File: UI/Web/src/app/on-deck/on-deck.component.ts [hunk 1]
```diff
@@ -63,10 +63,10 @@ export class OnDeckComponent implements OnInit {
     this.loadPage();
   }
 
-  updateFilter(data: SeriesFilter, isFirst) {
-    this.filter = data;
+  updateFilter(event: FilterEvent) {
+    this.filter = event.filter;
     const page = this.getPage();
-    if (page === undefined || page === null || isFirst !== true) {
+    if (page === undefined || page === null || event.isFirst !== true) {
```
Comment (majora2007) [inline UI/Web/src/app/on-deck/on-deck.component.ts:69]: We can just change this to !event.isFirst as it's a boolean.

**Review by majora2007 — CHANGES_REQUESTED** [review]
Almost there!

File: UI/Web/src/app/_models/series-filter.ts [hunk 2]
```diff
@@ -68,4 +68,9 @@ export const mangaFormatFilters = [
...
       selected: false
     }
-];
\ No newline at end of file
+];
+
+export interface FilterEvent {
+  filter: SeriesFilter;
+  isFirst: boolean;
+}
```
[the remark this answers is not in the document]
Comment (therobbiedavis) [reply UI/Web/src/app/_models/series-filter.ts]: resolved with 7ab4709

File: UI/Web/src/app/cards/card-detail-layout/card-detail-layout.component.ts [hunk 3]
```diff
@@ -572,11 +571,7 @@ export class CardDetailLayoutComponent implements OnInit, OnDestroy {
   }
 
   apply(isFirst) {
-    if (isFirst === true) {
-      this.applyFirstFilter.emit(this.filter);
-    } else {
-      this.applyFilter.emit(this.filter);
-    }
+    this.applyFilter.emit({filter: this.filter, isFirst: isFirst});
```
[the remark this answers is not in the document]
Comment (therobbiedavis) [reply UI/Web/src/app/cards/card-detail-layout/card-detail-layout.component.ts:575]: resolved with 7ab4709

File: UI/Web/src/app/on-deck/on-deck.component.ts (hunk 1 above)
Comment (therobbiedavis) [reply UI/Web/src/app/on-deck/on-deck.component.ts]: resolved with 7ab4709

**Review by majora2007 — APPROVED** [review]
We good to merge. Thanks for taking care of the comments.

**Pull request merged**